src/misc/helper.ts

```diff
@@ -0,0 +1,142 @@
+import { Transaction } from '../jitcoin/block';
+import { BlockHeader, TransactionElement, BlockBody } from './interfaces';
+
+/**
+ *
+ * @author Flo Dörr
+ * @param {string} previousBlockHash
+ * @param {string} merkleTree
+ * @param {number} nonce
+ * @param {Transaction[]} transactions
+ */
+export const saveBinaryHex = (
+  previousBlockHash: string,
+  merkleTree: string,
+  nonce: number,
+  transactions: Transaction[],
+) => {
+
+  createDir();
+
+  const file = getJitCoinFile();
+
+  // write magic bytes
+  const delimiterHex = Buffer.from(DELIMITER).toString('hex');
+
+  // write header
+  const header = {
+    version: VERSION,
+    previousBlockHash,
+    merkleTree,
+    nonce,
+    time: new Date().getTime(),
+  } as BlockHeader;
+
+  // write header
+  const headerHex = Buffer.from(stringify(header)).toString('hex');
+
+  // write transaction count
+  const lengthHex = Buffer.from(transactions.length.toString()).toString(
+    'hex',
+  );
+
+  const trans = new Array<string>();
+
+  for (const transaction of transactions) {
+    trans.push(
+      Buffer.from(stringify({
+        amount: transaction.amount,
+        randomHash: transaction.randomHash,
+        userId: transaction.userId,
+      } as TransactionElement)).toString('utf8'),
+    );
+  }
+
+  // write body
+  const body = {
+    transactions: trans,
+  } as BlockBody;
+
+  // write body
+  const bodyHex = Buffer.from(stringify(body)).toString('hex');
+
+  const sizeHex = Buffer.from(
+    '' + (headerHex.length + lengthHex.length + bodyHex.length),
+  ).toString('hex');
+
+  const data = delimiterHex + sizeHex + headerHex + lengthHex + bodyHex;
+
+  appendFileSync(file, data, { encoding: 'hex' });
+};
+
+/**
+ *
+ * @author Flo Dörr
+ * @param {string} filename
+ * @returns {number} filesize in bytes
+ */
+const getFilesize = (filename: string): number => {
+  const stats = statSync(filename);
+  const fileSizeInBytes = stats.size;
+  return fileSizeInBytes;
+};
+
+/**
+ *
+ * @author Flo Dörr
+ */
+const createDir = () => {
+  if(!existsSync(JITCOIN_DIR)){
+    mkdirSync(JITCOIN_DIR);
+  }
+  if(!existsSync(BLOCKCHAIN_DIR)){
+    mkdirSync(BLOCKCHAIN_DIR);
+  }
+};
+
+/**
+ *
+ * @author Flo Dörr
+ * @returns {string} current file
+ */
+const getJitCoinFile = (): string => {
+  const files = readdirSync(BLOCKCHAIN_DIR);
+  if(files.length === 0){
+    const file = BLOCKCHAIN_DIR + '/' + JITCOIN_FILE.replace('$', appendZeros(0));
+    writeFileSync(file, '');
+    return(file);
+  }
+  const currentFile = BLOCKCHAIN_DIR + '/' + files[files.length - 1];
+  if(getFilesize(currentFile) <= MAX_FILE_SIZE){
+    return currentFile;
+  }else{
+    const file = BLOCKCHAIN_DIR + '/' + JITCOIN_FILE.replace('$', appendZeros(files.length));
+    writeFileSync(file, '');
+    return file;
+  }
+};
+
+/**
+ *
+ * @author Flo Dörr
+ * @param {number} nmbr
+ * @returns {string} name of file
+ */
+const appendZeros = (nmbr: number): string => {
```

Comment:
I'm not sure if this is what you're trying to do here, but you could use [`padEnd`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd) for this task if it's just about zeropadding a string.

Reply:
Ohh I didn't know about this. I'll check on that one later.

Reply:
I cannot access padEnd although I added `"lib": [ "es2017"  ],` to the tsconfig.json. Any ideas on that?

Reply:
That's strange, can you try bumping up the target to use `esnext`? That should definitely work, and we're using the latest and greatest features anyway.

Reply:
@BrunoScheufler that one worked! Thanks!

Reply:
shuld be fixed with 12bbcd0